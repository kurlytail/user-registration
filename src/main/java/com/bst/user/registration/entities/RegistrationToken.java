package com.bst.user.registration.entities;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class RegistrationToken {
	
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
    private Date createdDate = new Date();
    @Column(unique=true)
    private String email;
    
    public RegistrationToken() {
    	this.email = null;
    }
    
    public RegistrationToken(String email) {
    	this.email = email;
    }

	public Long getId() {
		return id;
	}

	public Date getCreatedDate() {
		return createdDate;
	}

	public String getEmail() {
		return email;
	}
}
