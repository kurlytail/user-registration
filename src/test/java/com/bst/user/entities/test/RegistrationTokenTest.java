package com.bst.user.entities.test;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
import static org.powermock.api.mockito.PowerMockito.whenNew;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import com.bst.user.registration.entities.RegistrationToken;

@RunWith(PowerMockRunner.class)
@PrepareForTest(RegistrationToken.class)
public class RegistrationTokenTest {

	@BeforeClass
	public static void startSnapshot() throws Exception {
		start();
	}

	@AfterClass
	public static void stopSnapshot() throws Exception {
		validateSnapshots();
	}
	
	@Before
	public void setupDate() throws Exception {
	    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
	    sdf.setTimeZone(TimeZone.getTimeZone("PST"));

	    Date NOW = sdf.parse("2015-05-23 00:00:00");
 
	    whenNew(Date.class).withNoArguments().thenReturn(NOW);
	}

	@Test
	public void testRegistrationToken() {
		RegistrationToken token = new RegistrationToken();
		expect(token).toMatchSnapshot();
	}

	@Test
	public void testRegistrationTokenString() {
		RegistrationToken token = new RegistrationToken("some-email");
		expect(token).toMatchSnapshot();
	}

	@Test
	public void testGetEmail() {
		RegistrationToken token = new RegistrationToken("some-email");
		expect(token.getEmail()).toMatchSnapshot();
	}

}
