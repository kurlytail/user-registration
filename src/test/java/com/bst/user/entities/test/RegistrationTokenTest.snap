com.bst.user.registration.entities.test.RegistrationTokenTest.testGetEmail=[
  "some-email"
]


com.bst.user.registration.entities.test.RegistrationTokenTest.testRegistrationToken=[
  {
    "createdDate": "May 23, 2015 2:00:00 AM"
  }
]


com.bst.user.registration.entities.test.RegistrationTokenTest.testRegistrationTokenString=[
  {
    "createdDate": "May 23, 2015 2:00:00 AM",
    "email": "some-email"
  }
]